set.mm

```diff
@@ -263326,6 +263428,73 @@ irreducible elements (see ~ df-irred ).  (Contributed by Mario Carneiro,
       UHFGVKVNVOWJWLVCHIVKWDWEWNWPVCVDVE $.
   $}
 
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Subrings generated by a subset
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $( Introduce new constant symbols. $)
+  $c RingSpan $.
+
+  $( Extend class notation with span of a set of elements over a ring. $)
+  crgspn $a class RingSpan $.
+
+  ${
+    $d w s t $.
+    $( The ring-span of a set of elements in a ring is the smallest subring
+       which contains all of them.  (Contributed by Stefan O'Rear,
+       7-Dec-2014.) $)
+    df-rgspn $a |- RingSpan = ( w e. _V |-> ( s e. ~P ( Base ` w ) |->
+        |^| { t e. ( SubRing ` w ) | s C_ t } ) ) $.
+  $}
+
+  ${
+    $d ph a b t $.  $d R a b t $.  $d B a b t $.  $d A a b t $.  $d S t $.
+    rgspnval.r $e |- ( ph -> R e. Ring ) $.
+    rgspnval.b $e |- ( ph -> B = ( Base ` R ) ) $.
+    rgspnval.ss $e |- ( ph -> A C_ B ) $.
+    rgspnval.n $e |- ( ph -> N = ( RingSpan ` R ) ) $.
+    rgspnval.sp $e |- ( ph -> U = ( N ` A ) ) $.
+    $( Value of the ring-span of a set of elements in a ring.  (Contributed by
+       Stefan O'Rear, 7-Dec-2014.) $)
+    rgspnval $p |- ( ph -> U = |^| { t e. ( SubRing ` R ) | A C_ t } ) $=
+      ( vb va cfv wss cbs wcel cvv wceq crgspn csubrg crab cint fveq1d cpw cmpt
+      crg elex fveq2 pweqd rabeq syl inteqd mpteq12dv df-rgspn fvex mptex fvmpt
+      cv pwex 3syl eqid sseq1 rabbidv sseqtrd elpw2 sylibr wrex subrgid eqeltrd
+      sseq2 rspcev syl2anc intexrab sylib fvmptd3 eqtrd 3eqtrd ) AFCGOCEUAOZOZC
+      BUTZPZBEUBOZUCZUDZLACGVTKUEAWACMEQOZUFZMUTZWBPZBWDUCZUDZUGZOWFACVTWMAEUHR
+      ZESRVTWMTHEUHUINEMNUTZQOZUFZWJBWOUBOZUCZUDZUGWMSUAWOETZMWQWTWHWLXAWPWGWOE
+      QUJUKXAWSWKXAWRWDTWSWKTWOEUBUJWJBWRWDULUMUNUONBMUPMWHWLWGEQUQZVAURUSVBUEA
+      MCWLWFWHWMSWMVCWICTZWKWEXCWJWCBWDWICWBVDVEUNACWGPCWHRACDWGJIVFCWGXBVGVHAW
+      CBWDVIZWFSRADWDRCDPZXDADWGWDIAWNWGWDRHWGEWGVCVJUMVKJWCXEBDWDWBDCVLVMVNWCB
+      WDVOVPVQVRVS $.
+
+    $( The ring-span of a set is a subring.  (Contributed by Stefan O'Rear,
+       7-Dec-2014.) $)
+    rgspncl $p |- ( ph -> U e. ( SubRing ` R ) ) $=
+      ( vt cv wss csubrg cfv crab cint wcel eqeltrd rgspnval c0 wne ssrab2 wrex
+      cbs crg eqid subrgid sseq2 rspcev syl2anc rabn0 sylibr subrgint sylancr
+      syl ) AEBLMZNZLDOPZQZRZUTALBCDEFGHIJKUAAVAUTNVAUBUCZVBUTSUSLUTUDAUSLUTUEZ
+      VCACUTSBCNZVDACDUFPZUTHADUGSVFUTSGVFDVFUHUIUQTIUSVELCUTURCBUJUKULUSLUTUMU
+      NDVAUOUPT $.
+
+    $( The ring-span of a set contains the set.  (Contributed by Stefan O'Rear,
+       30-Nov-2014.) $)
+    rgspnssid $p |- ( ph -> A C_ U ) $=
+      ( vt cv wss csubrg cfv crab cint ssintub rgspnval sseqtrrid ) ABLMNLDOPZQ
+      RBELBUBSALBCDEFGHIJKTUA $.
+
+    rgspnmin.sr $e |- ( ph -> S e. ( SubRing ` R ) ) $.
+    rgspnmin.ss $e |- ( ph -> A C_ S ) $.
+    $( The ring-span is contained in all subspaces which contain all the
```

Comment:
Perhaps "subspaces" should be replaced by "subrings".

Reply:
Probably a copy-paste error by Stefan, I'll fix it in the next commit.